File: abc/abc132/abc132_b.cs
// https://atcoder.jp/contests/abc132/submissions/8272759

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var P = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var result = 0;
            for (var i = 1; i < N - 1; i++)
            {
                if (P[i - 1] < P[i] && P[i] < P[i + 1]
                    || P[i + 1] < P[i] && P[i] < P[i - 1])
                {
                    result++;
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc132/abc132_b.fs
// https://atcoder.jp/contests/abc132/submissions/11967721

let n = stdin.ReadLine() |> int
let p = stdin.ReadLine().Split() |> Array.map int

let rec solver s i =
    if i = n - 1 then s
    elif p.[i - 1] < p.[i] && p.[i] < p.[i + 1] || p.[i + 1] < p.[i] && p.[i] < p.[i - 1] then solver (s + 1) (i + 1)
    else solver s (i + 1)

solver 0 1 |> stdout.WriteLine
